locations/storefinders/store_locator_plus_self.py

```diff
@@ -0,0 +1,75 @@
+from scrapy import Spider
+from scrapy.http import FormRequest
+
+from locations.dict_parser import DictParser
+from locations.geo import point_locations
+
+# This store finder is a self-hosted WordPress plugin with a website
+# of https://wordpress.org/plugins/store-locator-le/ and source code
+# at https://github.com/wp-plugins/store-locator-le
+#
+# This store finder is not to be confused with the
+# software-as-a-service store locator of the same name, from the
+# same company, that is documented at https://storelocatorplus.com/
+#
+# To use this spider, specify a value for allowed_domains and the
+# default path for the StoreLocatorPlus API endpoint will be used.
+# In the event the default path is different, you can alternatively
+# specify a value for start_url. Then add one or more searchable
+# points files as searchable_points_files = [x, y..] and also
+# specify a non-zero number for search_radius = x. A non-zero number
+# is also required for the max_results attribute. This number is set
+# by the server and cannot be changed. To obtain this max_results
+# number, observe an API call and check the JSON response:
+# data_queries["standard_location_load"]["query"] to obtain the
+# number from the SQL LIMIT clause.
+#
+# An exception will be raised if max_results (or more) locations
+# are returned in any given radius search, as this indicates some
+# locations have been truncated. If this occurs, more granular
+# searchable points files need to be supplied, and search_radius
+# needs to be selected carefully to ensure that max_results (or
+# more) locations are never returned for any radius search.
+#
+# If clean ups or additional field extraction is required from the
+# source data, override the parse_item function. Two parameters are
+# passed, item (and ATP "Feature" class) and location (a dict which
+# is returned from the store locator JSON response for a particular
+# location).
+
+
+class StoreLocatorPlusSelfSpider(Spider):
+    searchable_points_files = []
+    search_radius = 0
+    max_results = 0
+
+    def start_requests(self):
+        if hasattr(self, "allowed_domains"):
+            url = f"https://{self.allowed_domains[0]}/wp-admin/admin-ajax.php"
+        else:
+            url = self.start_urls[0]
+        if url and len(self.searchable_points_files) > 0 and self.search_radius != 0 and self.max_results != 0:
+            for searchable_points_file in self.searchable_points_files:
+                for lat, lon in point_locations(searchable_points_file):
+                    formdata = {
+                        "action": "csl_ajax_onload",
+                        "lat": str(lat),
+                        "lng": str(lon),
+                        "radius": str(self.search_radius),
+                    }
+                    yield FormRequest(url=url, formdata=formdata, method="POST")
```

Comment:
FYI Wordpress can do url parameters. I think it is slightly better as the logs are more helpful.
https://www.hemmakvall.se/wp-admin/admin-ajax.php?action=csl_ajax_onload

Reply:
I just tried it out and I don't think this is true for at least StoreLocatorPlusSelf. For example:

`https://rulerfoods.com/wp-admin/admin-ajax.php?action=csl_ajax_onload&lat=42.4255736&lng=-67.8072665&radius=500`

The response to that HTTP GET query shows that StoreLocatorPlus is not able to determine the coordinates or search radius from parameters supplied in the URL.

Reply:
Hmm, maybe some plugins don't do it. TIL.